src/lib/sanity/queries/fragments/fragment.types.ts

```diff
@@ -0,0 +1,30 @@
+import { GetPageQueryResult, PostQueryResult, PostsArchiveQueryResult } from '@/sanity.types';
+
+export type PostCardFragmentType = NonNullable<PostsArchiveQueryResult['results'][number]>;
```

Comment:
Could this one come from `PostQueryResult`?

Reply:
No, the PostQueryResult uses the full post data (Post Fragment) while th archive uses the PostCard Fragment.

I'm introducing this pattern in this PR because right now we don't distinguish between where data is used. Wherever we are using a post (post page, listings, cards on the homepage, etc.) we are usng the same data from the same fragment.
This will likely result in overfetching data resulting in more data being sent to the client, more complex queries resulting in slower query time, etc.
E.g. for cards we don't need the full richtext editor, not the SEO data, etc. So we should therefore introduce a pattern that aims for querying only (about) that data that we need.